neuraxle/steps/numpy.py

```diff
@@ -51,6 +51,19 @@ def transform_one(self, data_inputs):
     def _concat(self, data_inputs):
         return np.concatenate(data_inputs, axis=-1)
 
+class NumpyConcatenateOnTimeStep(NonFittableMixin, BaseStep):
```

Comment:
For the `NumpyConcatenateOuterBatch` for example, people could have knew without documentation that `Outer` refers to the outermost axis and that it was the batch size, but here it may be unclear unless a) docstrings are added or b) that the class be instead be named NumpyConcatenateOnTimeStepAxis1 instead. I'd prefer docstrings, the other class will also need some docstrings eventually. Perhaps that by inheriting from the same base concatenator that the only docstrings to be written many times are 1) the classes-level docstring and 2) the constructor method docstring of the classes.